spec/validators/date_not_in_future_validator_spec.rb

```diff
@@ -0,0 +1,36 @@
+require "rails_helper"
+
+RSpec.describe DateNotInFutureValidator do
+  subject do
+    Class.new {
+      include ActiveModel::Validations
+      attr_accessor :date
+      validates :date, date_not_in_future: true
+
+      def self.name
+        "Transaction"
+      end
+    }.new
+  end
+
+  describe "when date is nil" do
+    it "is valid" do
+      subject.date = nil
+      expect(subject.valid?).to be true
+    end
+  end
+
```

Comment:
I think having a test that checks what the behaviour is when the date is _today_ would be good to have. We'd expect it to be true, I think?